sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/SchemaPruning.scala

```diff
@@ -164,7 +164,12 @@ object SchemaPruning extends Rule[LogicalPlan] {
       outputRelation: LogicalRelation,
       prunedBaseRelation: HadoopFsRelation) = {
     val prunedOutput = getPrunedOutput(outputRelation.output, prunedBaseRelation.schema)
-    outputRelation.copy(relation = prunedBaseRelation, output = prunedOutput)
+    // also add the metadata output if any
+    // TODO: should be able to prune the metadata schema
```

Comment:
hmm, seems like column pruning is automatically supported if we remove the code change here?

Reply:
actually no:
1. this rule supports only: `Parquet` + `ORC` and controls by a flag `spark.sql.optimizer.nestedSchemaPruning.enabled`
2. this rule only prunes the data schema (comparing the data schema with the project + filter)

I am thinking:
- refactor this rule to prune metadata
OR
- make a new rule `MetadataSchemaPruning`
OR
- prune the metadata at `FileSourceStrategy` (might be hacky)

Reply:
makes sense. Let's add a new rule `MetadataSchemaPruning` later.

Reply:
sure! thanks